opal/corelib/string.rb

```diff
@@ -1575,7 +1575,7 @@ def upto(stop, excl = false, &block)
     return enum_for :upto, stop, excl unless block_given?
     stop = Opal.coerce_to(stop, String, :to_str)
     %x{
-      var a, b, s = self.toString();
+      var a, b, s = self.toString(), value;
```

Comment:
Yes, `break` is a language construct, it breaks out of the call site with the block lexically containing the `break`
(like an invisible exception implicitly caught by the call site with the block):
```ruby
ret = foo.bla(1,2,3) { break 42 if rand < 0.5 }
# execution continues here
```
would break of the full call to `#bla` and `ret` would be assigned the value passed to break.
This should be in language specs, and maybe it makes sense to add a comment to specs using `break` in core to clarify it's not method-specific (or alternatively remove them since it's not method specific).
Sorry for the confusion, we had the same problem in JRuby+Truffle as well.